Hmm, not yet:
```
[   31s] [ 92%] Generating syntax/vim/generated.vim
[   31s] cd /home/abuild/rpmbuild/BUILD/neovim/build/runtime && /usr/bin/luajit /home/abuild/rpmbuild/BU
ILD/neovim/src/nvim/generators/preload.lua /home/abuild/rpmbuild/BUILD/neovim /home/abuild/rpmbuild/BUIL
D/neovim/build/lib/libnlua0.so /home/abuild/rpmbuild/BUILD/neovim/scripts/genvimvim.lua /home/abuild/rpm
build/BUILD/neovim/src/nvim /home/abuild/rpmbuild/BUILD/neovim/build/runtime/syntax/vim/generated.vim /h
ome/abuild/rpmbuild/BUILD/neovim/build/funcs_data.mpack
[   31s] /usr/bin/luajit: attempt to index a userdata value
[   31s] stack traceback:
[   31s]        [C]: at 0x7f967cc49250
[   31s]        [C]: in function 'require'
[   31s]        /home/abuild/rpmbuild/BUILD/neovim/scripts/genvimvim.lua:1: in main chunk
[   31s]        [C]: at 0x558888a9e110
[   31s] make[2]: *** [runtime/CMakeFiles/runtime.dir/build.make:96: runtime/syntax/vim/generated.vim] E
rror 1
[   31s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/neovim/build'
[   31s] make[2]: *** Waiting for unfinished jobs....
```
That’s on openSUSE/Tumbleweed with lua51-mpack-1.0.6-5.2 and the current master.

--- a/scripts/genvimvim.lua
+++ b/scripts/genvimvim.lua
@@ -1,4 +1,4 @@
-local mpack = require('mpack')
+local mpack = vim.mpack
 
 if arg[1] == '--help' then
   print('Usage: lua genvimvim.lua src/nvim runtime/syntax/vim/generated.vim')

--- a/src/mpack/lmpack.c
+++ b/src/mpack/lmpack.c
@@ -1171,9 +1171,6 @@ static const luaL_reg mpack_functions[] = {
   {NULL, NULL}
 };
 
-#ifdef NLUA_WIN32
-  __declspec(dllexport)
-#endif
 int luaopen_mpack(lua_State *L)
 {
   /* Unpacker */

--- a/src/mpack/lmpack.h
+++ b/src/mpack/lmpack.h
@@ -1,6 +1,3 @@
 #include <lua.h>
 
-#ifdef NLUA_WIN32
-  __declspec(dllexport)
-#endif
 int luaopen_mpack(lua_State *L);

--- a/src/nlua0.c
+++ b/src/nlua0.c
@@ -1,0 +1,13 @@
+#include <lua.h>
+
+#include "mpack/lmpack.h"
+
+LUA_API int luaopen_nlua0(lua_State* L);
+
+LUA_API int luaopen_nlua0(lua_State* L) {
+  lua_getglobal(L, "vim");
+  luaopen_mpack(L);
+  lua_setfield(L, -2, "mpack");
+
+  return 1;
+}

--- a/src/nvim/CMakeLists.txt
+++ b/src/nvim/CMakeLists.txt
@@ -15,7 +15,7 @@ endif()
 
 add_library(nlua0 MODULE)
 if(WIN32)
-  target_compile_definitions(nlua0 PUBLIC LUA_BUILD_AS_DLL NLUA_WIN32)
+  target_compile_definitions(nlua0 PUBLIC LUA_BUILD_AS_DLL LUA_LIB)
   set_target_properties(nlua0 PROPERTIES ENABLE_EXPORTS TRUE)
 elseif(APPLE)
   set_target_properties(nlua0 PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
@@ -322,6 +322,7 @@ if(PREFER_LUA)
   target_compile_definitions(main_lib INTERFACE NVIM_VENDOR_BIT)
 endif()
 
+list(APPEND NLUA0_SOURCES ${PROJECT_SOURCE_DIR}/src/nlua0.c)
 foreach(subdir
         os
         api

--- a/src/nvim/generators/gen_api_dispatch.lua
+++ b/src/nvim/generators/gen_api_dispatch.lua
@@ -1,4 +1,4 @@
-local mpack = require('mpack')
+local mpack = vim.mpack
 
 local hashy = require'generators.hashy'
 

--- a/src/nvim/generators/gen_api_ui_events.lua
+++ b/src/nvim/generators/gen_api_ui_events.lua
@@ -1,4 +1,4 @@
-local mpack = require('mpack')
+local mpack = vim.mpack
 
 assert(#arg == 5)
 local input = io.open(arg[1], 'rb')

--- a/src/nvim/generators/gen_eval.lua
+++ b/src/nvim/generators/gen_eval.lua
@@ -1,4 +1,4 @@
-local mpack = require('mpack')
+local mpack = vim.mpack
 
 local autodir = arg[1]
 local metadata_file = arg[2]

--- a/src/nvim/generators/preload.lua
+++ b/src/nvim/generators/preload.lua
@@ -4,6 +4,7 @@ package.path = srcdir .. '/src/nvim/?.lua;' ..srcdir .. '/runtime/lua/?.lua;' ..
 _G.vim = require'vim.shared'
 _G.vim.inspect = require 'vim.inspect'
 package.cpath = package.cpath .. ';' .. nlualib
+require 'nlua0'
 
 arg[0] = table.remove(arg, 1)
 return loadfile(arg[0])()

